tools/releases/version.sh

```diff
@@ -26,6 +26,31 @@ function envoy_version() {
   set +o nounset
 }
 
+function tools_version() {
+  set -o errexit
+  set -o pipefail
+  set -o nounset
+  # The command finds the most recent tag that is reachable from a commit
+  describe=$(git describe --tags 2>/dev/null || echo "none")
+  currentBranch=$(git rev-parse --abbrev-ref HEAD 2> /dev/null || echo "no-branch")
+  # On master the closest tag is 2.0.0 so we are setting dev for master
+  if [[ ${describe} == "2.0.0"* ]]
```

Comment:
Got it: it's not 2.1.0 because of: https://kongstrong.slack.com/archives/C7J8S9N6Q/p1677255829164009 (squash merge instead of merge commit when merging release-2.1 into master) :)

Reply:
not sure how to differentiate master from the latest release branch.

Reply:
I've tried `^[1-9]+.[0-9]+.0` but I found that if branch had no patches it's going to point to dev config

Reply:
I've introduced something different now:  I check if the newest tag on the branch `^[1-9]+.[0-9]+.0` matches the pattern, and later I check if the newest tag on the branch starts with the 2 first elements of the newest tag in the repository. It doesn't work correctly on the master, because of the wrong merge, but if we push the tag correctly it should be good.